<launch>
    <!-- Static Transform Publisher -->
    
    <arg name="config" default="$(find storm_ros)/config/config_robot_world_publisher.yaml" doc="Path to parameter config file"/>

    <include file="$(find realsense2_camera)/launch/rs_camera.launch"> </include>     

    <!-- RealSense Camera Node
    <node pkg="realsense2_camera" type="realsense2_camera_node" name="realsense_camera">
        <param name="align_depth" type="bool" value="true"/>
        <param name="enable_sync" type="bool" value="true"/>
    </node> -->

    <!-- ArUco Marker Detection Node -->
    <!-- fill in the following parameters according to your tracking system -->
    <arg name="markerId"        default="23"/>      <!-- set your marker id -->
    <arg name="markerSize"      default="0.05"/>    <!-- in m -->
    <arg name="eye"             default="left"/>
    <arg name="marker_frame"    default="aruco_marker_frame"/> 
    <arg name="ref_frame"       default=""/>  <!-- leave empty and the pose will be published wrt param parent_name -->
    <arg name="corner_refinement" default="LINES" /> <!-- NONE, HARRIS, LINES, SUBPIX -->    
    <arg name="camera_frame" default="camera_color_optical_frame" />
    <arg name="camera_image_topic" default="/camera/color/image_raw" />
    <arg name="camera_info_topic" default="/camera/color/camera_info" /> 

    <node pkg="aruco_ros" type="single" name="aruco_single">
        <remap to="$(arg camera_info_topic)" from="/camera_info" />
        <remap to="$(arg camera_image_topic)" from="/image" />
        <param name="image_is_rectified" value="True"/>
        <param name="marker_size"        value="$(arg markerSize)"/>
        <param name="marker_id"          value="$(arg markerId)"/>
        <param name="reference_frame"    value="$(arg ref_frame)"/>   <!-- frame in which the marker pose will be refered -->
        <param name="camera_frame"       value="$(arg camera_frame)"/>
        <param name="marker_frame"       value="$(arg marker_frame)" />
        <param name="corner_refinement"  value="$(arg corner_refinement)" />
    </node>
    
    <node pkg="tf2_ros" type="static_transform_publisher" name="camera_broadcaster"
    args="0.9830213929111877 -0.012718213930554917 0.9634871525263239 -0.6741867304453285 -0.6812170892723831 0.2004538230286437 0.20306106127936216 base_link camera_color_optical_frame" />

    <node name="robot_world_pub" pkg="storm_ros" type="robot_world_publisher.py" output="screen" clear_params="true">
        <rosparam file="$(arg config)" />
    </node>
    
    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find storm_ros)/rviz/robot_world_visualizer.rviz" />

</launch>
